src/Microsoft.Extensions.Logging/Logger.cs

```diff
@@ -136,6 +136,44 @@ public IDisposable BeginScope<TState>(TState state)
             return scope;
         }
 
+        public void RecordMetric(Metric metric)
+        {
+            var loggers = Loggers;
+            if (loggers == null)
+            {
+                return;
+            }
+
+            List<Exception> exceptions = null;
+            foreach (var loggerInfo in loggers)
+            {
+                // REVIEW: What "level" should metrics be at?
```

Comment:
Also whould we be able to define metric filters?

Reply:
Well, the level comment isn't really relevant in the new design, but yeah, we probably should support metric filters. The pattern of `DefineMetric(string)` can do that by returning a single-instance "black-hole" metric recorder.